### DISCUSSION TITLE
why Fast-CDR but not protobuf?

### DISCUSSION THREAD

**Discussion by chengfang0911** [eProsima/Fast-DDS#3847] [Q&A]
why Fast-DDS use Fast-CDR to serialization but not protobuf?

**Comment by JLBuenoLopez** [comment]
Fast DDS is an implementation of the [OMG DDS specification](https://www.omg.org/spec/DDS/1.4/About-DDS). The DDS Interoperability Wire Protocol is defined in the [DDSI-RTPS specification](https://www.omg.org/spec/DDSI-RTPS/2.3/About-DDSI-RTPS). This specification defines the Common Data Representation (CDR) to be used for DDS interoperability. Fast CDR is an implementation of Section 10 of the specification. Consequently, protobuf is not used because the specification states that CDR must be used in order to be interoperable with other DDS implementations. Nevertheless, you can prepare your own TypeSupport so you can use Fast DDS with protobuf serialization.

**Reply by chengfang0911** [reply]
thanks
